### ISSUE TITLE
InvalidCastException on creating migration with entity with explicit interface implementation

### ISSUE THREAD

**Issue by HKaresz** [dotnet/efcore#14458]
I have several entities which has an enum-typed primary key (like the Country entity in the attached sample). These tables are code tables, their data is not supposed to change, only with data seeding. Until 2.2.0-preview3-35497 creating migrations worked well, but after upgrading to >= 2.2.0, I cannot create new migration, I got the following exception:

```
Exception message:
System.InvalidCastException: Invalid cast from 'System.Int32' to 'ConsoleApp1.Countries'.
Stack trace:
   at System.Convert.DefaultToType(IConvertible value, Type targetType, IFormatProvider provider)
   at Microsoft.EntityFrameworkCore.Storage.ValueConversion.ValueConverter`2.Sanitize[T](Object value)
   at Microsoft.EntityFrameworkCore.Storage.ValueConversion.ValueConverter`2.<>c__DisplayClass3_0`2.<SanitizeConverter>b__0(Object v)
   at Microsoft.EntityFrameworkCore.Metadata.Internal.EntityType.GetData(Boolean providerValues)
   at Microsoft.EntityFrameworkCore.Migrations.Design.CSharpSnapshotGenerator.GenerateEntityType(String builderName, IEntityType entityType, IndentedStringBuilder stringBuilder)
   at Microsoft.EntityFrameworkCore.Migrations.Design.CSharpSnapshotGenerator.GenerateEntityTypes(String builderName, IReadOnlyList`1 entityTypes, IndentedStringBuilder stringBuilder)
   at Microsoft.EntityFrameworkCore.Migrations.Design.CSharpSnapshotGenerator.Generate(String builderName, IModel model, IndentedStringBuilder stringBuilder)
   at Microsoft.EntityFrameworkCore.Migrations.Design.CSharpMigrationsGenerator.GenerateMetadata(String migrationNamespace, Type contextType, String migrationName, String migrationId, IModel targetModel)
   at Microsoft.EntityFrameworkCore.Migrations.Design.MigrationsScaffolder.ScaffoldMigration(String migrationName, String rootNamespace, String subNamespace, String language)
   at Microsoft.EntityFrameworkCore.Design.Internal.MigrationsOperations.AddMigration(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.AddMigrationImpl(String name, String outputDir, String contextType)
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.<>c__DisplayClass3_0`1.<Execute>b__0()
   at Microsoft.EntityFrameworkCore.Design.OperationExecutor.OperationBase.Execute(Action action)
Invalid cast from 'System.Int32' to 'ConsoleApp1.Countries'.
```

### Steps to reproduce
The attached project generates the migration properly.
After upgrading Microsoft.EntityFrameworkCore package to a stable 2.2.x version and generate a new migration, I got the exception above.
[ConsoleApp1.zip](https://github.com/aspnet/EntityFrameworkCore/files/2772850/ConsoleApp1.zip)

### Further technical details
EF Core version: 2.2.0
Database Provider: Microsoft.EntityFrameworkCore.SqlServer
Operating system: Windows 10 x64
IDE: Visual Studio 2017 15.9

**Comment by ajcvickers** [comment]
Notes for triage: This only fails creating the model snapshot. Built model seems fine (see below) and EnsureCreated works and seeds correctly.

Code:
```C#
public enum Countries
{
    Country1 = 1,
    Country2 = 2,
    Country3 = 3
}

public interface IEntity
{
    int? Id { get; }
}

public class Country : IEntity
{
    public Countries? Id { get; protected internal set; }

    public string Name { get; protected internal set; }

    int? IEntity.Id => (int?)Id;
}

public class TestContext : DbContext
{
    protected override void OnConfiguring(DbContextOptionsBuilder optionsBuilder)
    {
        optionsBuilder.UseSqlServer(@"Server=(localdb)\mssqllocaldb;Database=Test;ConnectRetryCount=0");
    }

    protected override void OnModelCreating(ModelBuilder modelBuilder)
    {
        base.OnModelCreating(modelBuilder);

        modelBuilder.Entity<Country>(b =>
        {
            b.HasKey(x => x.Id);
            b.Property(x => x.Name)
                .IsRequired()
                .HasMaxLength(50);

            b.HasData(new[]
            {
                new Country { Id = Countries.Country1, Name = "Country1" },
                new Country { Id = Countries.Country2, Name = "Country2" },
                new Country { Id = Countries.Country3, Name = "Country3" },
            });
        });
    }
}
```
Model:
```
Model: 
  EntityType: Country
    Properties: 
      Id (Nullable<Countries>) Required PK AfterSave:Throw 0 0 0 -1 -1
      Name (string) Required MaxLength50 1 1 -1 -1 -1
    Keys: 
      Id PK
```

**Comment by lajones** [comment]
Repros on 3.1.2, but does not repro using latest build from master.

**Comment by lajones** [comment]
**EF Team Triage:** We think this is probably fixed in our working code base. Would you be able to try our nightly builds and see if the issue is resolved for you. Nightly builds are available from https://www.myget.org/Gallery/aspnetcidev.

*BTW this is a canned response and may have info or details that do not directly apply to this particular issue. While we'd like to spend the time to uniquely address every incoming issue, we get a lot traffic on the EF projects and that is not practical. To ensure we maximize the time we have to work on fixing bugs, implementing new features, etc. we use canned responses for common triage decisions.*

**Comment by lajones** [comment]
Note: apparently the canned response above is inaccurate. Instructions for the nightly builds are [here](https://github.com/dotnet/aspnetcore/blob/master/docs/DailyBuilds.md). 

Looks like this got fixed as part of the fix for #19495. So closing.

**Comment by ajcvickers** [comment]
@lajones Duplicates don't go in a milestone or have a type. (Avoids double counting.)
